### ISSUE TITLE
Bucket notification not URL encoded object key correctly

### ISSUE THREAD

**Issue by xiang-zhu** [localstack/localstack#2741]
# Type of request: This is a ...

[x] bug report
[ ] feature request

# Detailed description
When using bucket notification through localstack, the key in the notification message is not url encoded as AWS specification 

This behavior seems to be broken since v0.10.7, v0.10.6 is the last working version
...

## Expected behavior
https://docs.aws.amazon.com/AmazonS3/latest/dev/notification-content-structure.html
...

## Actual behavior
key not encoded
...

# Steps to reproduce

## Command used to start LocalStack

...

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

...

**Comment by usmangani1** [comment]
Hi @xiang-zhu .Thanks for reporting the issue.Can you please help us with a small example? A small aws cli command ,actual and expected response of the command will help us understand the issue in bit detail and help us to resolve the issue sooner.Thanks.

**Comment by xiang-zhu** [comment]
Hi, here's a code snippet that can demonstrate the bug.

```
#!/usr/bin/env bash                                                                                                                                                                                                 
set -e                                                                                                                                                                                                              
                                                                                                                                                                                                                    
export AWS_ACCESS_KEY_ID=x                                                                                                                                                                                          
export AWS_SECRET_ACCESS_KEY=x                                                                                                                                                                                      
export AWS_DEFAULT_REGION=us-east-1                                                                                                                                                                                 
                                                                                                                                                                                                                    
docker run --rm -p 4566:4566 -e SERVICES=s3,sqs -d --name bucket-notificaton-bug localstack/localstack:latest > /dev/null                                                                                           
sleep 5                                                                                                                                                                                                             
echo "Started Localstack with Docker"                                                                                                                                                                               
                                                                                                                                                                                                                    
aws s3api --endpoint-url http://localhost:4566 create-bucket --bucket test-bucket > /dev/null                                                                                                                       
aws sqs --endpoint-url http://localhost:4566 create-queue --queue-name test-queue > /dev/null                                                                                                                       aws s3api --endpoint-url http://localhost:4566 put-bucket-notification-configuration --bucket test-bucket --notification-configuration '{"QueueConfigurations": [{"QueueArn": "arn:aws:sns:us-east-1:000000000000:test-queue","Events": ["s3:ObjectCreated:*"]}]}' > /dev/null                                                                                                                                                                                                                                                                                                                                                                              echo "test" > text.txt                                                                                                                                                                                              aws s3 --endpoint-url http://localhost:4566 cp text.txt s3://test-bucket/a@b > /dev/null                                                                                                                            rm -f text.txt                                                                                                                                                                                                      
                                                                                                                                                                                                                    
aws sqs --endpoint-url http://localhost:4566 receive-message --queue-url http://localhost:4566/000000000000/test-queue                                                                                              
                                                                                                                                                                                                                    
docker stop bucket-notificaton-bug > /dev/null                                                                                                                                                                      
echo "Stopped Localstack with Docker" 
```

The received message looks like this

```
{
    "Messages": [
        {
            "MessageId": "a004328e-5b85-f0a7-3a78-67595dc569f9",
            "ReceiptHandle": "cjyommycxumlttrjfkklvntfaamugdcrpuibdaumdjlhxoryfvddjagupnrftmswdybnqqbbivsicjwlpowcstnykfjhrvdfkbjoefcmnlfcnmezyqfvrwbnabyvzcrqakqlrqbihopedzkkbuobmaxmddicafjghgyntzbbdjadzwnwtujxanewy",
            "MD5OfBody": "edfd9b871a6e8570a27c3258e73f9572",
            "Body": "{\"Records\": [{\"eventVersion\": \"2.0\", \"eventSource\": \"aws:s3\", \"awsRegion\": \"us-east-1\", \"eventTime\": \"2020-07-18T22:41:57.029Z\", \"eventName\": \"ObjectCreated:Put\", \"userIdentity\": {\"principalId\": \"AIDAJDPLRKLG7UEXAMPLE\"}, \"requestParameters\": {\"sourceIPAddress\": \"127.0.0.1\"}, \"responseElements\": {\"x-amz-request-id\": \"bc4d798e\", \"x-amz-id-2\": \"eftixk72aD6Ap51TnqcoF8eFidJG9Z/2\"}, \"s3\": {\"s3SchemaVersion\": \"1.0\", \"configurationId\": \"testConfigRule\", \"bucket\": {\"name\": \"test-bucket\", \"ownerIdentity\": {\"principalId\": \"A3NL1KOZZKExample\"}, \"arn\": \"arn:aws:s3:::test-bucket\"}, \"object\": {\"key\": \"a@b\", \"size\": 5, \"eTag\": \"d41d8cd98f00b204e9800998ecf8427e\", \"versionId\": null, \"sequencer\": \"0055AED6DCD90281E5\"}}}]}",
            "Attributes": {
                "SenderId": "AIDAIT2UOQQY3AUEKVGXU",
                "SentTimestamp": "1595112117060",
                "ApproximateReceiveCount": "1",
                "ApproximateFirstReceiveTimestamp": "1595112117399"
            }
        }
    ]
}
```

The object key in the notification is `a@b`, which should have been url encoded to `a%40b`

**Comment by whummer** [comment]
Thanks for the update @xiang-zhu @usmangani1 . This should be fixed in #2847, and the change should be available in the `latest` version of the Docker image. Please report here if the problem persists. Thanks
